src/model.rs

```diff
@@ -2,40 +2,40 @@ use crate::{
     error::Result,
     util::{get_type_name, hash},
 };
-use serde::{Deserialize, Serialize, Serializer};
-use std::{
-    collections::BTreeMap,
-    path::{Path, PathBuf},
-    result,
-};
-
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{collections::BTreeMap, path::PathBuf, result};
+/// Converts a model instance into a consistent yaml.
 ///
 /// # Errors
-/// Error out with ``serde_yaml`` seralization error if something goes wrong
-pub fn to_yaml<T: Serialize>(item: &T) -> Result<String> {
-    let mut yaml = serde_yaml::to_string(item)?;
+///
+/// Will return `Err` if there is an issue converting an `instance` into YAML (w/o annotation).
+pub fn to_yaml<T: Serialize>(instance: &T) -> Result<String> {
+    let mut yaml = serde_yaml::to_string(instance)?;
     yaml.insert_str(0, &format!("class: {}\n", get_type_name::<T>())); // replace class at top
+
     Ok(yaml)
 }
 
+// --- core model structs ---
+
 /// A reusable, containerized computational unit.
-#[derive(Serialize, Deserialize, Debug, PartialEq, Default)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Default, Clone)]
 pub struct Pod {
     /// Metadata that doesn't affect reproducibility.
     #[serde(skip)]
     pub annotation: Option<Annotation>,
-    command: String,
     /// Unique id based on reproducibility.
     #[serde(skip)]
     pub hash: String,
     image: String,
+    command: String,
     input_stream_map: BTreeMap<String, StreamInfo>,
     output_dir: PathBuf,
     output_stream_map: BTreeMap<String, StreamInfo>,
+    source_commit_url: String,
```

Comment:
- Leave this later for discussion
Maybe change it to point to the repo, (github honors renaming forever???)